Sources/SwiftMemcache/Extensions/ByteBuffer+SwiftMemcache.swift

```diff
@@ -64,12 +75,29 @@ extension ByteBuffer {
     ///
     /// - returns: A `MemcachedFlags` instance populated with the flags read from the buffer.
     mutating func readMemcachedFlags() -> MemcachedFlags {
-        let flags = MemcachedFlags()
+        var flags = MemcachedFlags()
+        var ttlIsUnlimited = false
         while let nextByte = self.getInteger(at: self.readerIndex, as: UInt8.self) {
             switch nextByte {
             case UInt8.whitespace:
                 self.moveReaderIndex(forwardBy: 1)
                 continue
+            case UInt8.t:
+                self.moveReaderIndex(forwardBy: 1)
+                if let currentByte = self.getInteger(at: self.readerIndex, as: UInt8.self), currentByte == UInt8.hyphen {
+                    ttlIsUnlimited = true
+                    self.moveReaderIndex(forwardBy: 1)
+                }
+                if let ttl: UInt32 = self.readIntegerFromASCII() {
+                    if ttlIsUnlimited {
+                        // If TTL is negative, set it as indefinite.
+                        flags.timeToLive = nil
+                        ttlIsUnlimited = false
```

Comment:
How do we differentiate the case when TTL is `nil` because it is not present or when it is nil because it is infinite? I think we should probably store this a bit differently in the `MemcacheFlags` struct

Reply:
In our `MemcacheFlags` we can explore storing timeToLive as our TimeToLive enum (instead of as a `UInt32`), that way when TTL is present and indefinite we store it as `flags.timeToLive = .indefinitely` and when TTL is present and has a expiry time we store it as `flags.timeToLive = .expiresAt(instant)`

Reply:
That sounds good. We should probably store it as a `TimeToLive?`